src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/ConsoleParameters.cs

```diff
@@ -40,9 +40,22 @@ public ConsoleParameters(IFileHelper fileHelper)
     }
 
     /// <summary>
-    /// Environment variables to be set for the process
+    /// Environment variables to be set for the process. This will add the specified entries to the environment variables
+    /// inherited from the current process. If you wish to provide a full set of environment variables yourself set <see cref="ClearEnvironmentVariables"/> to true.
     /// </summary>
-    public Dictionary<string, string> EnvironmentVariables { get; set; }
+    public Dictionary<string, string> EnvironmentVariables { get; set; } = new Dictionary<string, string>();
+
+    /// <summary>
+    /// Clears all environment variables that would be inherited from machine, user and process
+    /// and only sets the entries you provided in <see cref="EnvironmentVariables"/>.
+    /// This allows you to provide an arbitrary set of environment variables, for example when you
+    /// want to skip environment variables that are set in the IDE that starts vstest.console.
+    /// When setting this to true you are responsible for providing a full set of envirionment variables
+    /// that allow the process to start.
+    /// You most likely want to use <see cref="System.Environment.GetEnvironmentVariables(System.EnvironmentVariableTarget)"/> and combine
+    /// <see cref="System.EnvironmentVariableTarget.Machine"/> and <see cref="System.EnvironmentVariableTarget.User"/> responses.
+    /// </summary>
+    public bool ClearEnvironmentVariables { get; set; }
```

Comment:
What about `SkipInheritedEnvironmentVariables`? Or `UseOnlyProvidedEnvironmentVariables`?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.environment?view=net-6.0 docs

Reply:
@nohwnd The challenge with the "ProcessInfo method" is that in Test Explorer the customer expects us to duplicate the ENV that VS is running in more or less exactly, as if the test host is a child process of VS. So it would require a lot more code to add/modify/delete keys to make the environments match. That is why we originally went with the solution of just setting all the variables.

I think `ClearEnvironmentVariables` is pretty clear, and this looks like a good improvement on the existing API for sure. It won't be hard to fixup Test Explorer to handle this. The only downside to this that I can see is that it could make it difficult to remove specific env variables. But you can always just set them to an empty string, so that might not even be an issue.

Reply:
@peterwald  you are already providing us with the full set of environment variables that you want to use. We just won't clear them for you by default anymore. So with the ProcessStartInfo like approach, you would need to call Clear() on the EnvironmentVariableDictionary, and then put all your key value pairs into it, e.g. by using .ForEach. It is a tiny bit of wasted performance, but you don't start vstest.console that often.

I created PR this morning on your repo it is <10 lines. Using the ProcessStartInfo approach will be the same.

Reply:
I think we already expected that the vstest process would inherit the environment variables from the main process and that we only needed to add extra environment variables we needed, so I think we won't use `ClearEnvironmentVariables`. It is probably an improvement in our case that environment variables will no longer be cleared by default. 

That said `ClearEnvironmentVariables` seems like an implementation bleed through the API. What the user of the API wants is functionally something like `DisableInheritEnvironmentVariables` or something like that.

Reply:
Actually we use a custom testhostlauncher and don't use ConsoleParameters EnvironmentVariables so I'm not sure this case is relevant to us at all: https://github.com/stryker-mutator/stryker-net/blob/master/src/Stryker.Core/Stryker.Core/TestRunners/VsTest/StrykerVstestHostLauncher.cs#L36

Reply:
@rouke-broersma IIRC you set dotnet roll-forward in the current process to allow vstest.console to inherit it, and run on any dotnet version available. 

This change would have no impact on you other than being able to set the env variables directly on vstest.console rather than setting them in your process and letting vstest.console inherit them.

Reply:
I used `InheritEnvironmentVariables` and set it to `true` by default. Seems like the least complicated way to name that option. 

I kept the option approach, to avoid copying the whole environment into the dictionary, only to have VS clear it and copy almost whole environment into the dictionary again.

Anyone else gets the simplest behavior they can. They give us a dictionary of env variables they want set, and we merge them with their current environment. They don't need to be editing an existing dictionary, and possibly get it wrong by using `.Add(key, value)` instead of `[key] = value`.